pre_commit_hooks/remove_eol_characters.py

```diff
@@ -0,0 +1,49 @@
+#! /usr/bin/env python
+
+import codecs
+import re
+import sys
+from typing import List, Tuple
+
+
+def remove_eol() -> Tuple[str, str]:
+    eol_character = r'\n'
+    # \u4e00-\u9fff contains all chinese characters
+    characters = r'\u4e00-\u9fff'
+    # Unicode halfwidth and fullwidth forms, refer to
+    # https://en.wikipedia.org/wiki/Halfwidth_and_Fullwidth_Forms_(Unicode_block)
+    punctuations = r'\uff01-\uff9f'
+    # Find natural Chinese paragraphs that are split by end_of_line characters.
+    # The pattern is: Chinese characters/punctuations with one and only one
+    # end_of_line in between.
+    pattern = fr'([{characters}{punctuations}]){eol_character}([{characters}])'
+    # This replacement will remove the end_of_line character in between
+    repl = r'\1\2'
+    return pattern, repl
+
+
+def rewrite_file(file_name: str, strategies: List[Tuple[str, str]]) -> bool:
+    with codecs.open(file_name, mode='r', encoding='utf-8') as f:
```

Comment:
Can we replace `codes.open` with `open` because `codes.open` also calls `open` internally. More details at https://github.com/python/cpython/blob/83886261fa584b45d5e70c5a422f69cad7422a0a/Lib/codecs.py#L905